L1Trigger/GlobalTrigger/plugins/L1GTPrescaler.cc

```diff
@@ -0,0 +1,97 @@
+#include <vector>
+#include <array>
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/one/EDFilter.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "DataFormats/L1GlobalTrigger/interface/L1GlobalTriggerReadoutRecord.h"
+
+class L1GTPrescaler : public edm::one::EDFilter<> {
+public:
+  L1GTPrescaler(edm::ParameterSet const& config);
+
+  virtual bool filter(edm::Event& event, edm::EventSetup const& setup) override;
+
+  static  void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
+
+private:
+  const edm::EDGetTokenT<L1GlobalTriggerReadoutRecord> m_l1ResultsToken;
+  const std::vector<double> m_algoPrescales;
+  const std::vector<double> m_techPrescales;
+  std::vector<unsigned int> m_algoCounters;
+  std::vector<unsigned int> m_techCounters;
```

Comment:
hi David,
one issue I have with std::array<T,N> is that it cannot be easily constructed from an std::vector<T>, as returned by config.getParameter<std::vector<double>>(...) :-(